src/modules/flow/iio/iio.json

```diff
@@ -822,6 +822,12 @@
              "name": "sampling_frequency"
          },
          {
+             "data_type": "string",
+             "default": null,
+             "description": "Channel name. If none, try to auto detect channel",
+             "name": "channel_name"
```

Comment:
Yeah, it will call device reader callback twice. That's one more thing that needs to be coordinated. 
A simple approach would use a list of reader callbacks per device, instead of one. But then, a single tick will trigger all nodes, is this a problem? (In your test case, it would trigger 4 reads each time.) So, each device added would add to a list of callbacks, and the whole list is called by sol-iio.c.
Other ideas?